cogs/stockmarket/market.py

```diff
@@ -0,0 +1,568 @@
+
+    # Simple Information
+    @market.command(name="info")
+    async def get_info(self, ctx, ticker: str):
+        """Gets information about a particular stock"""
+        ticker = ticker.upper()
+        info = yf.Ticker(ticker).info
+        print(info)
+
+        embed = await self.build_embed(ctx, ceil(info['regularMarketPrice'] * 100), ticker, info)
+        short_name = info['shortName']
+        embed.set_footer(text=f'{ticker} - {short_name}')
+        try:
+            summary = info['longBusinessSummary']
+            summary = (summary[:1021] + '...') if len(summary) > 1021 else summary
+            embed.add_field(name=f'About {ticker}', value=summary, inline=False)
+        except KeyError:
+            embed.add_field(name=f'About {ticker}',
+                            value=f'No information found for ticker {ticker}',
+                            inline=False)
+
+        await ctx.send(embed=embed)
+
+    # Simple Information
+    @market.command(name="transactions")
+    async def get_sinfo(self, ctx, num_results: int=10):
+        """lists the last num_results transactions for a member"""
+
+        async with self.config.member(ctx.author).transactions() as transactions:
+            if not transactions:
+                await ctx.send(f'{ctx.author.mention} you have not made any transactions!')
+                return
+
+            if num_results > len(transactions):
+                num_results = len(transactions)
+
+            msg = f'{ctx.author.mention} your last {num_results} transactions:\n'
+            msg += '```css\n'
+            for transaction in transactions[-num_results:]:
+                type_transaction = transaction['type']
+                ticker = transaction['ticker']
+                date_str = transaction['date']
+
+                if type_transaction == 'buy':
+                    msg += f'Bought   [{ticker:<8}] [{transaction["shares_bought"]:<8} shares] for {transaction["bought_price"]:<8} each @ {date_str}\n'
+                elif type_transaction == 'sell':
+                    msg += f'Sold     [{ticker:<8}] [{transaction["shares_sold"]:<8} shares] for {transaction["sold_price"]:<8} each {date_str}\n'
+                else:
+                    if transaction["dividend_amount"] != 0:
+                        msg += f'Dividend [{ticker:<8}] [{transaction["dividend_amount"]:<8}] @ {date_str}\n'
+                    if transaction["split_ratio"] != 0:
+                        msg += f'Split   [{ticker:<8}] [{transaction["split_ratio"]:<8}] @ {date_str}\n'
+
+            msg += '```'
+
+            await ctx.send(msg)
+
+    async def get_embed_chart(self, ticker, period_str='1d', interval_str='5m'):
+        """Gets the chart of the requested stock"""
+        plt.clf()
+        ticker = ticker.upper()
+        # Get the data.. takes like 0.2s
+        data = yf.download(ticker, period=period_str, interval=interval_str)

```

Comment:
Here and everywhere we use `yf`, are these blocking call? If so, this could potentially stall the bot.

Reply:
would taking the `yf` library, and converting it all to async/await functions help unblock?